proto/storage/report_snapshot.proto

```diff
@@ -25,12 +25,16 @@ message ReportSnapshot {
   ReportType type = 5;
   oneof filter {
     VulnerabilityReportFilters vuln_report_filters = 6;
+    ViewBasedVulnerabilityReportFilters view_based_vuln_report_filters = 12;
   }
   CollectionSnapshot collection = 7;
   storage.Schedule schedule = 8;
   ReportStatus report_status = 9;
   repeated NotifierSnapshot notifiers = 10;
   SlimUser requester = 11;
+  //fields related to ondemand reports
+  string view_based_report_request_name = 13;
+  string area_of_concern = 14;
```

Comment:
Should this be an enum?

Reply:
No this should be a string. in future we will let users define views so we cant limit this to a enum